src/client/datascience/jupyter/kernels/cellExecutionJob.ts

```
@@ -0,0 +1,189 @@
+        while (this.queueOfCellsToExecute.length) {
+            // Take the first item from the queue, this way we maintain order of cell executions.
+            const cellToExecute = this.queueOfCellsToExecute[0];
+            if (!cellToExecute) {
+                continue;
+            }
+            traceCellMessage(cellToExecute.cell, 'Before Execute individual cell');
+            const executionResult = await this.executeIndividualCell(cellToExecute, notebook);
+            traceCellMessage(cellToExecute.cell, `After Execute individual cell ${executionResult}`);
+            // If a cell has failed the get out.
+            if (
+                this.cancelledOrCompletedWithErrors ||
+                executionResult === vscodeNotebookEnums.NotebookCellRunState.Error
+            ) {
+                this.cancelledOrCompletedWithErrors = true;
+                traceInfo(`Cancel all remaining cells ${this.cancelledOrCompletedWithErrors} || ${executionResult}`);
+                await this.cancel();
+                break;
+            }
+            this.onCellExecutionCompleted(cellToExecute);
+        }
+    }
+    private async executeIndividualCell(
+        cellExecution: CellExecution,
+        notebook: INotebook
+    ): Promise<NotebookCellRunState | undefined> {
+        traceCellMessage(cellExecution.cell, 'Push cell into queue for execution');
+        const chainedExecution = this.chainedCellExecutionPromise.then(async () => {
+            traceCellMessage(cellExecution.cell, 'Get cell from queue for execution');
+            // Start execution
+            await cellExecution.start(notebook);
+
+            // The result promise will resolve when complete.
+            const promise = cellExecution.result;
+            promise
+                .finally(() => traceCellMessage(cellExecution.cell, 'Cell from queue completed execution'))
+                .catch(noop);
```

Why would exceptions be swallowed?

---

We return the original promise and don't swallow those.
I'm adding a finally for logging (so the original promise is still left untouched).
Will simplify this, i was just guarding against traceCellMesage failing & unnecessarily causing other errors, hut `traceCellMessage` cannot fail.

---

Oh yeah duh. You're not setting the promise to the catch result.